### PR TITLE
fix(server): better fix for the OAuth Discovery errors

### PR DESCRIPTION
**Pull request by samip5**

### CODE REVIEW

**Comment by jrasm91** [comment]
Probably yes with your tsconfig change

**Comment by samip5** [comment]
I have fixed it since @jrasm91

File: server/tsconfig.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
       "@app/infra/*": ["src/infra/*"],
       "@app/domain": ["src/domain"],
       "@app/domain/*": ["src/domain/*"]
-    }
+    },
```
Comment (jrasm91) [inline server/tsconfig.json:29]:
```suggestion
    }
```

File: server/tsconfig.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
     "experimentalDecorators": true,
     "allowSyntheticDefaultImports": true,
     "resolveJsonModule": true,
-    "target": "es2017",
+    "target": "es2021",
```
Comment (jrasm91) [inline server/tsconfig.json:11]: Why did you change this? Does the type not exist in es2017 or something?

File: server/src/domain/auth/auth.service.ts [hunk 3]
```diff
@@ -339,8 +339,12 @@ export class AuthService {
...
 
       return new issuer.Client(metadata);
-    } catch (error: Error | any) {
-      this.logger.error(`Error in OAuth discovery: ${error}`, error?.stack);
+    } catch (error: any) {
+      if (error instanceof AggregateError) {
+        this.logger.error(`Error in OAuth discovery: ${error}`, error?.stack, error?.errors);
+      } else if (error instanceof Error) {
+        this.logger.error(`Error in OAuth discovery: ${error}`, error?.stack);
+      }
```
Comment (jrasm91) [inline server/src/domain/auth/auth.service.ts:347]: I believe you can just change line 342 to be `error: any | AggregateError` and then use a single logger.error with `error?.errors`, instead of if/else with copy/pasted log lines.

File: server/tsconfig.json (hunk 2 above)
Comment (samip5) [reply server/tsconfig.json:11]: Correct, it doesn't exist until 2021.

File: server/src/domain/auth/auth.service.ts (hunk 3 above)
Comment (samip5) [reply server/src/domain/auth/auth.service.ts:347]: Original problem was that it wouldn't even find the AggregateError, but now it does, so fixed it to be like that instead.

**Review by jrasm91 — APPROVED** [review]
